### PR TITLE
fix: validate genesis

### LINKED ISSUE #2435: Inconsistent genesis after network reset
**Describe the bug**
After betanet reset, my account which was previously validator still has locked amount but is not a validator.

**To Reproduce**
1. Stake 100,000 N
2. Get network reset via betanet release process
3. Observe locked amount on the non-validating account:

E.g. 
```
near state illia
Account illia
{
  amount: '109999999898346855000000000',
  locked: '161748960602985038993033587691',
  code_hash: '11111111111111111111111111111111',
  storage_usage: 510,
  storage_paid_at: 0,
  block_height: 2109542,
  block_hash: 'BJcXpqf2zvL98XE1r2gqKH66qxcjtHr3uJt8UqGoRBZt',
  formattedAmount: '109.999999898346855'
}
```

**Expected behavior**
Locked amount should be reset to 0 on network reset if we are reseting `validators` et.

Also we should have a hard failure on inconsistent genesis at the launch between `validators` and `locked` amounts.

**Version (please complete the following information):**
Live betanet

### PR DESCRIPTION
**Pull request by bowenwang1996**
Validate genesis to check that various invariants hold before starting the node. Partially addresses #2435.

Test plan
---------
unit tests in `validate_genesis.rs`.

### CODE REVIEW

File: neard/src/genesis_validate.rs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    #[test]
+    #[should_panic(expected = "validator accounts do not match staked accounts")]
+    fn test_validator_not_match() {
+        let mut genesis = Genesis::default();
+        genesis.config.validators = vec![AccountInfo {
+            account_id: "test".to_string(),
+            public_key: PublicKey::empty(KeyType::ED25519),
+            amount: 100,
+        }];
+        genesis.config.total_supply = 110;
```
Comment (ilblackdragon) [inline neard/src/genesis_validate.rs:109]: Don't we compute total_supply when we load genesis from records in the first place?

File: neard/src/genesis_validate.rs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    let mut contract_account_ids = HashSet::new();
+    for record in genesis.records.0.iter() {
+        match record {
+            StateRecord::Account { account_id, account } => {
+                if account_ids.contains(account_id) {
+                    panic!("Duplicate account id {} in genesis records", account_id);
+                }
+                total_supply += account.locked + account.amount;
+                account_ids.insert(account_id.clone());
+                if account.locked > 0 {
```
Comment (ilblackdragon) [inline neard/src/genesis_validate.rs:34]: btw, it's totally possible if we stop mid epoch, that account is locked but didn't make it to validators. I'm assuming we are fixing this somewhere - but where?

File: neard/src/main.rs [hunk 3]
```diff
@@ -146,6 +147,7 @@ fn main() {
         ("run", Some(args)) => {
             // Load configs from home.
             let mut near_config = load_config(home_dir);
+            validate_genesis(&near_config.genesis);
```
Comment (ilblackdragon) [inline neard/src/main.rs:150]: This is good, but similar to above question - do we update genesis on the dump, removing extra locked / etc?
We do need to add more complex stuff into `state_migration` to catch such situation - for example send a staking transaction which would make `locked > 0` but wouldn't add to validators list.

Comment (bowenwang1996) [reply neard/src/main.rs:150]: yes will address that in a separate PR.

File: neard/src/genesis_validate.rs (hunk 1 above)
Comment (bowenwang1996) [reply neard/src/genesis_validate.rs:109]: doesn't hurt to double check

File: neard/src/genesis_validate.rs (hunk 2 above)
Comment (bowenwang1996) [reply neard/src/genesis_validate.rs:34]: will address in a separate PR

**Pull request merged**